src/client.cpp

```diff
@@ -762,7 +766,12 @@ void CClient::Init()
     const int iPrefMonoFrameSize = iSndCrdPrefFrameSizeFactor * SYSTEM_FRAME_SIZE_SAMPLES;
 
     // get actual sound card buffer size using preferred size
-    iMonoBlockSizeSam = Sound.Init ( iPrefMonoFrameSize );
+    const int iSystemMonoSizeSam = Sound.Init ( iPrefMonoFrameSize );
+
+    // prevent any early audio callbacks during configuration
```

Comment:
That should never happen. The Init() function is only called if the Sound.Stop() was called before. Can you please give me more information in which case you saw a crash here?

Reply:
I think the JACK backend doesn't have any stop function implemented. It just lazily sets a flag, while yet the function may still be processing. I found this when working on another patch, where I could change the buffersizes used by Jamulus from the GUI when using JACK audio backend, and it core-dumped over and over again :-(

Reply:
In CSound::process there is a check implemented: pSound->IsRunning(). If pSound->IsRunning() is false, pSound->ProcessCallback is not called. So there is already a protection implemented. We have to find out why this protection did not work for your case. Can you please give more information of what you did when it crashed? If you say "I found this when working on another patch", maybe you have changed something in the Jamulus code so that it crashed.

Reply:
I still think that "MutexAudioProcess" is not necessary for the current Jamulus source code on the Git master branch.

Reply:
The problem is not new audio callbacks, but what if an audio callback is already pending when doing init? Then the check is ineffective.

Reply:
At leas the JACK driver does not drain the audio callback, and if there is a callback process, then there is just simple sleep of 5 ms.

Reply:
I fixed this in the sound base class instead like you suggested. When we stop() then aquire the new mutex, and drop it. After that the callback should always see the correct value of the running flag! See updated patch.